apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorReportingMatching.cs

```diff
@@ -0,0 +1,133 @@
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ErrorReporting.V1Beta1;
+using Google.Protobuf.WellKnownTypes;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Cloud.Diagnostics.Common.Tests
+{
+    internal class ErrorReportingMatching
+    {
+        internal const int ConflictStatusCode = 409;
+        internal const string ServiceName = "SomeService";
+        internal const string VersionName = "1.0.0";        
+        internal const string UserAgentValue = "user-agent-1.0";
+        internal static readonly Uri GoogleUri = new Uri("https://www.google.com");
+        internal static readonly string DeleteMethod = "DELETE";
+        internal static readonly string ExceptionMessage = "some exception message";
+
+        private readonly string _defaultUri;
+        private readonly string _defaultMethod;
+        private readonly bool _isWindows;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ErrorReportingMatching"/> used to help matching
+        /// Error Reporting Unit tests.
+        /// </summary>
+        /// <param name="isWindows">True if this test is running on windows.</param>
+        /// <param name="defaultUri">The default uri expected for an http request uri.
+        ///     Defaults to the empty string.</param>
+        /// <param name="defaultMethod">The default method expected for an http request method.
+        ///     Defaults to the empty string.</param>
+        internal ErrorReportingMatching(bool isWindows, string defaultUri = "", string defaultMethod = "")
+        {
+            _isWindows = isWindows;
+            _defaultUri = defaultUri;
+            _defaultMethod = defaultMethod;
+        }
+
+        /// <summary>
+        /// Matcher to check if an <see cref="IEnumerable{ReportedErrorEvent}"/> contains
+        /// one element and matches the following:
+        ///   - The EventTime less than or equal to now
+        ///   - The HttpMethod matches the default http method
+        ///   - The Url matches the default url
+        ///   - The UserAgent is empty or null
+        ///   - The ResponseStatusCode is "0"
+        ///   - The LineNumber is "0"
+        ///   - The FilePath is empty or null
+        ///   - The FunctionName is empty or null
+        ///   - The Service is <see cref="ServiceName"/>
+        ///   - The Version is <see cref="VersionName"/>
+        /// </summary>
```

Comment:
Not important here since this isn't public, but just for future reference, making bulleted lists like this won't work when this gets turned into docs. I think the dashes will just wrap like any other character. You'd need to do something like this instead:

```csharp
/// <summary>
/// Something:
/// <list type="bullet">
/// <term>A</term>
/// <term>B</term>
/// <term>C</term>
/// </list>
/// </summary>
```